protostar/commands/test/test_case_runners/setup_case_runner.py

```diff
@@ -0,0 +1,39 @@
+from protostar.commands.test.environments.setup_case_execution_environment import (
+    SetupCaseExecutionEnvironment,
+)
+from protostar.commands.test.starkware.test_execution_state import TestExecutionState
+from protostar.commands.test.test_environment_exceptions import ReportedException
+from protostar.commands.test.test_results import (
+    SetupCaseResult,
+    PassedSetupCaseResult,
+    FailedSetupCaseResult,
+)
+from protostar.commands.test.test_suite import TestCase
+
+
+async def run_setup_case(
+    test_case: TestCase, state: TestExecutionState
+) -> SetupCaseResult:
```

Comment:
[INFO] I'll probably make this function a class to inject `SetupCaseExecutionEnvironment` at some point in the future. 

My vision on the testing module:
- `TestRunner.worker` instantiates `TestRunnerBuilder/Factory`
- `TestRunnerBuilder/Factory` acts as a `composition_root`
- The core testing logic is moved out of `TestCommand` to `protostar.test_runner` component

Reply:
Beware that `TestExecutionState` is sensitive to forks, and it will have to be passed carefully.